Packs/Graylog/Integrations/Graylog/Graylog.py

```diff
@@ -0,0 +1,158 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+''' IMPORTS '''
+
+from datetime import datetime
+
+import dateparser
+import requests
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+
+def results_return(command, thingtoreturn):
+    results = CommandResults(
+        outputs_prefix='Graylog.' + str(command),
+        outputs_key_field='',
+        outputs=thingtoreturn
+    )
+    return_results(results)
+
+
+def test_module(client):
+    result = client._http_request('GET', 'cluster/')
+    if result:
+        return 'ok'
+    else:
+        return 'Test failed: ' + str(result)
+
+
+def create_incident_from_log(log):
+    occurred = log['timestamp']
+    keys = log.keys()
+    labels = []
+    for key in keys:
+        labels.append({'type': key, 'value': str(log[key])})
+        formatted_description = 'Graylog Incident'
+    return {
+        'name': formatted_description,
+        'labels': labels,
+        'rawJSON': json.dumps(log),
+        'occurred': occurred
+    }
+
+
+def form_incindents(logs):
+    listofincidents = []
+    for item in logs:
+        listofincidents.append(create_incident_from_log(item['message']))
+    return listofincidents
+
+
+def fetch_incidents(client):
+    timefrom = dateparser.parse(demisto.params().get('fetch_time')).strftime('%Y-%m-%dT%H:%M:%S.%f')[:-3]
+    timefrom += 'Z'
+    incidentquery = demisto.params().get('fetch_query')
+    last_run = demisto.getLastRun()
+    if last_run and 'start_time' in last_run:
+        start_time = last_run.get('start_time')
+        start_time += 'Z'
+    else:
+        start_time = timefrom
+    end_time = datetime.now().strftime('%Y-%m-%dT%H:%M:%S.%f')[:-3]
+    end_time += 'Z'
+    parameters = {'query': incidentquery,
+                  'from': start_time,
+                  'to': end_time}
+    results = client._http_request('GET', '/search/universal/absolute', params=parameters)
+    if 'total_results' in results and results['total_results'] > 0:
+        demisto.setLastRun({'start_time': datetime.now().strftime('%Y-%m-%dT%H:%M:%S.%f')[:-3]})
```

Comment:
In order not to lose data in the time it takes to query the results please use `end_time` as the new `start_time` you are setting.

Reply:
Yeah, good point, a bit of a thinking error on my part there. Changed start_time to be the end_time for the next round.